hack/build-and-push.sh

```diff
@@ -1,10 +1,56 @@
 #!/bin/bash
 
+# There are two major types of tasks in build-definitions:
+#
+# - Normal tasks, which are written as Tekton Task resource.
+#
+# - kustomized tasks, which are customized by kustomization based on normal
+#   tasks. kustomized tasks can be customized either based on another normal
+#   task, e.g. task buildah-24gb is based on task buildah. This type of
+#   kustomized task inherits the interface without change. Or based on itself,
+#   e.g. task inspect image.
```

Comment:
"Based on itself" means there's a `$task_dir/$task_name.yaml` file _as well as_ a `$task_dir/kustomization.yaml`?

Technically, the code treats those as normal tasks I think

Reply:
No. As the example task `inspect-image` shows, version 0.2 is based on 0.1.

Reply:
Ooh, interesting. makes sense.